Fix issue warning message

juliusalberto opened the pull request:
<!-- 
Describe the changes you have made here: what, why, ... 
Link the issue that will be closed, e.g., "Closes #333".
If your PR closes a koppor issue, link it using its URL, e.g., "Closes https://github.com/koppor/jabref/issues/47".
"Closes" is a keyword GitHub uses to link PRs with issues; do not change it.
Don't reference an issue in the PR title because GitHub does not support auto-linking there.
-->

Ref #10121. Based on the suggestions of the repo maintainer I have:

1. Passes down the `NotificationService` to `CleanupWorker`. I called the notification service directly from the `CleanupWorker` but I'm not sure as I added error formatting in the `CleanupWorker`. Ideally, there will be another method in the `DialogService` class that can take a list of exceptions and print it in the dialog window.
2. Added strings in the `Jabref_en.properties` file

### Mandatory checks

<!-- 
- Go through the list below. Please don't remove any items.
- [x] done; [ ] not done / not applicable
-->

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

juliusalberto committed:
implement showing warning message in the cleanupworker

juliusalberto committed:
Edited changelog

juliusalberto committed:
Merge branch 'JabRef:main' into fix-issue-warning-message

JxCl-L commented:
Looks good to me! Your changes address the issue effectively. Nice work on implementation and updating the necessary files. Ready for merge in my opinion.

juliusalberto commented:
> Looks good to me! Your changes address the issue effectively. Nice work on implementation and updating the necessary files. Ready for merge in my opinion.

Do you think I should move the formatting logic in the `org.jabref.logic.cleanupworker` to somewhere in the `DialogService` class? I'm not sure about it - I really don't want to change the codebase too much. I think it's better to create a pull request - and then ask the maintainer how they want to proceed with it. What do you think? @JxCl-L

Noah-Martin1 commented:
Why are 4 of the checks failing?

juliusalberto committed:
Merge branch 'main' into fix-issue-warning-message

# Conflicts:
#	src/main/resources/l10n/JabRef_en.properties

juliusalberto committed:
Add missing Localization

juliusalberto committed:
Edited casting

koppor committed:
Test WIP version of action (#11989)

koppor committed:
Revert "Test WIP version of action (#11989)" (#11990)

koppor committed:
Switch priorities: XMP over Grobid (#11992)

* Switch priorities

* Add link to PR

koppor committed:
Streamline text of DownloadFullTextAction (#11993)

Co-authored-by: Christoph <siedlerkiller@gmail.com>

koppor committed:
Streamline code of org.jabref.logic.util.io.CitationKeyBasedFileFinder#findAssociatedFiles (#11994)

Co-authored-by: Christoph <siedlerkiller@gmail.com>

koppor committed:
Fix link in CONTRIBUTING.md (#11995)

koppor committed:
Fix typo (#12002)

leaf-soba committed:
remove Deprecated getText() (#11841)

* remove Deprecated getText()

remove Deprecated getText()

* Fix the code to pass unit test

Maybe I should add more unit test based on my change?

* Add more unit test to support my change

1. more unit test
2. rollback logic to follow old code which assumed that databaseContext is never null

* change to @ParameterizedTest test

change to @ParameterizedTest test

* update follow requested change

1. remove test prefix
2. remove unnecessary inline the helper method
3. fix a type
4. two minor refactor in both files.

* remove unused import

remove import org.jspecify.annotations.Nullable;

---------

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

koppor committed:
Ignore rsync errors on uploading (#12008)

Siedlerchr committed:
New Crowdin updates (#12007)

* New translations jabref_en.properties (French)

* New translations jabref_en.properties (Polish)

* New translations jabref_en.properties (Portuguese, Brazilian)

* New translations jabref_en.properties (Spanish)

* New translations jabref_en.properties (Arabic)

* New translations jabref_en.properties (Danish)

* New translations jabref_en.properties (German)

* New translations jabref_en.properties (Greek)

* New translations jabref_en.properties (Finnish)

* New translations jabref_en.properties (Italian)

* New translations jabref_en.properties (Japanese)

* New translations jabref_en.properties (Korean)

* New translations jabref_en.properties (Dutch)

* New translations jabref_en.properties (Norwegian)

* New translations jabref_en.properties (Portuguese)

* New translations jabref_en.properties (Russian)

* New translations jabref_en.properties (Swedish)

* New translations jabref_en.properties (Turkish)

* New translations jabref_en.properties (Ukrainian)

* New translations jabref_en.properties (Chinese Simplified)

* New translations jabref_en.properties (Chinese Traditional)

* New translations jabref_en.properties (Vietnamese)

* New translations jabref_en.properties (Indonesian)

* New translations jabref_en.properties (Persian)

* New translations jabref_en.properties (Tagalog)

koppor committed:
Update 0011-test-external-links-in-documentation.md (#12006)

koppor committed:
Output error if requirements tracing fails (#12003)

* Output error if requirements tracing fails

* Add word to dict

* Fix requirements chain

* Fix casing

Red12138-java committed:
Fix for #12004issue:Rewrite handleStringData to use CompositeIdFetcher (#12009)

* Rewrite handleStringData to use CompositeIdFetcher #12004
add a no-args constructor to CompositeIdFetcher.java

* delete the no-args constructor.

---------

Co-authored-by: u7744550 <u7744550@anu.edu.au>

koppor committed:
Add "After submission of a pull request" (#12010)

* Add "After submission of a pull request"

* Fix space

Siedlerchr committed:
Update to javafx 23.0.1 (#12013)

koppor committed:
Update takanome-dev/assign-issue-action to beta (#12012)

* Update takanome-dev/assign-issue-action to beta

* Update actions to test actions

* Add required "maitnainers" field

RabidGhost committed:
clarify After submission of a pull request in CONTRIBUTING.md

RabidGhost committed:
Update to MADR 4.0.0 (#12016)

* fix typos and update to MADR 4.0.0

* update to MADR 4.0.0

* improve readability

* remove redundant escape characters

koppor committed:
Fix removing comments (#12019)

koppor committed:
Try other template format (#12011)

* Try other template format

* Even shorter

adambash12 committed:
"Converted test cases in FileNameCleanerTest to use @CsvSource for im… (#12014)

* "Converted test cases in FileNameCleanerTest to use @CsvSource for improved readability. Added additional test cases for Linux to handle file name restrictions.

* Fixed checkstyle error

* Removed empytline for checkstyle

* Fixed CSV parsing issues by enclosing values containing commas in double quotes.

Test case was failing so had to fix

* Changed the custom test

* Fixed custom test

* Fixed custom test (again)

* Revised custom test

* Fixed the ordering of expected and input

* Swapped for all test cases

* Added removed test case

* Added forgotten comma

* Added singlequotes so csvSource can differentiate the 2 strings

RabidGhost committed:
clarify CONTRIBUTING.md

koppor committed:
Add label addition and removal of status: * (#12026)

koppor committed:
Use the beta version (#12027)

koppor committed:
Fix placeholders (#12028)

koppor committed:
Fix environement variable name for gh cli (#12029)

koppor committed:
Add some links into the code (#12030)

koppor committed:
Try to fix token and condition (#12031)

koppor committed:
Exhanges a TODO to a comment

Siedlerchr committed:
Merge pull request #12032 from JabRef/remove-TODO-by-comment

Exhanges a TODO to a comment

Siedlerchr committed:
Merge pull request #12023 from RabidGhost/update-CONTRIBUTING.md

Clarify CONTRIBUTING.md

trickypr committed:
Add SSRN urls to CompositeIdFetcher (#12021)

* feat: add ssrn urls to CompositeIdFetcher

* refactor: delete now-dead functions in ImportHandler

#12009 started using CompositeIdFetcher for fetching logic. This removes the dead code functions in ImportHandler

* docs: add ssrn import to changelog

* style: conform to stylecheck

* style: address requested changes

* docs: correct DOI capitalisation

---------

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

koppor committed:
Fix condition (#12041)

lllllllittlesun committed:
Allow selecting content of other user's comments(Fix for issue #11106) (#12020)

* Fix issue #11106, make others comment could be copied.

* Update CHANGELOG.md for fix issue #11106

* Modify code formatting

* Implement the method "setEditable" in "MarkdownEditor".
Adjust the order of the code

* Remove duplicate content in CHANGELOG.md

---------

Co-authored-by: Christoph <siedlerkiller@gmail.com>

koppor committed:
Add debug

koppor committed:
Try other token in on-review-submitted.yml

koppor committed:
Add required permissions

dependabot[bot] committed:
Bump org.itsallcode.openfasttrace from 3.0.0 to 3.0.1 (#12046)

Bumps org.itsallcode.openfasttrace from 3.0.0 to 3.0.1.

---
updated-dependencies:
- dependency-name: org.itsallcode.openfasttrace
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump com.dlsc.gemsfx:gemsfx from 2.58.0 to 2.60.0 (#12047)

Bumps [com.dlsc.gemsfx:gemsfx](https://github.com/dlsc-software-consulting-gmbh/GemsFX) from 2.58.0 to 2.60.0.
- [Release notes](https://github.com/dlsc-software-consulting-gmbh/GemsFX/releases)
- [Changelog](https://github.com/dlsc-software-consulting-gmbh/GemsFX/blob/master/jreleaser.yml)
- [Commits](https://github.com/dlsc-software-consulting-gmbh/GemsFX/compare/v2.58.0...v2.60.0)

---
updated-dependencies:
- dependency-name: com.dlsc.gemsfx:gemsfx
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.openrewrite.rewrite from 6.23.4 to 6.25.1 (#12048)

Bumps org.openrewrite.rewrite from 6.23.4 to 6.25.1.

---
updated-dependencies:
- dependency-name: org.openrewrite.rewrite
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.glassfish.jersey.inject:jersey-hk2 from 3.1.8 to 3.1.9 (#12050)

Bumps org.glassfish.jersey.inject:jersey-hk2 from 3.1.8 to 3.1.9.

---
updated-dependencies:
- dependency-name: org.glassfish.jersey.inject:jersey-hk2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.mockito:mockito-core from 5.14.1 to 5.14.2 (#12049)

Bumps [org.mockito:mockito-core](https://github.com/mockito/mockito) from 5.14.1 to 5.14.2.
- [Release notes](https://github.com/mockito/mockito/releases)
- [Commits](https://github.com/mockito/mockito/compare/v5.14.1...v5.14.2)

---
updated-dependencies:
- dependency-name: org.mockito:mockito-core
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump src/main/resources/csl-styles from `b413a77` to `5cfc7ae` (#12051)

Bumps [src/main/resources/csl-styles](https://github.com/citation-style-language/styles) from `b413a77` to `5cfc7ae`.
- [Release notes](https://github.com/citation-style-language/styles/releases)
- [Commits](https://github.com/citation-style-language/styles/compare/b413a778b8170cf5aebbb9aeffec62cfd068e19e...5cfc7ae46c1d596351d75aaa11ab06e98e0aeeaf)

---
updated-dependencies:
- dependency-name: src/main/resources/csl-styles
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump src/main/resources/csl-locales from `8bc2af1` to `4753e3a` (#12052)

Bumps [src/main/resources/csl-locales](https://github.com/citation-style-language/locales) from `8bc2af1` to `4753e3a`.
- [Release notes](https://github.com/citation-style-language/locales/releases)
- [Commits](https://github.com/citation-style-language/locales/compare/8bc2af16f5180a8e4fb591c2be916650f75bb8f6...4753e3a9aca4b806ac0e3036ed727d47bf8f678e)

---
updated-dependencies:
- dependency-name: src/main/resources/csl-locales
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump buildres/abbrv.jabref.org from `d870374` to `50edbd5` (#12053)

Bumps [buildres/abbrv.jabref.org](https://github.com/JabRef/abbrv.jabref.org) from `d870374` to `50edbd5`.
- [Release notes](https://github.com/JabRef/abbrv.jabref.org/releases)
- [Commits](https://github.com/JabRef/abbrv.jabref.org/compare/d87037495de7213b896dbb6a20170387de170709...50edbd56165d8efa27be7d06ff1df5d20aa8d2c7)

---
updated-dependencies:
- dependency-name: buildres/abbrv.jabref.org
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

koppor committed:
Replace `@deprecated` by explaining text (#12055)

juliusalberto committed:
Merge branch 'main' of https://github.com/JabRef/jabref into fix-issue-warning-message

juliusalberto committed:
Implemented the suggestion, moved the showFailure() to the GUI package

koppor committed:
Readd `@deprecated` annotation (#12056)

* Readd `@deprecated` annotation

* Fix casing

koppor committed:
Fix wrong permission

koppor committed:
Fix permissions (again)

koppor committed:
Try more permissions

koppor committed:
Debug: Does it fail because the label does not exist?

koppor committed:
Does not work - even with "permissions: write-all"

Refs https://github.com/cli/cli/issues/6274

LoayGhreeb committed:
Searching with Postgres (#11803)

* Convert to record

* Add embedded-postgres dependency

* Setup embedded postgre server

* Indexing entries with Postgres

* Remove Lucene bib fields indexer

* Use PreparedStatement to fix escaping characters

* Rename LuceneManager to IndexManager

* Begin to implement "new" search syntax

Co-authored-by: Loay Ghreeb <52158423+LoayGhreeb@users.noreply.github.com>
Co-authored-by: Carl Christian Snethlage <50491877+calixtus@users.noreply.github.com>

* Some new operators

* Change log level of EmbeddedPostgres

* Handle search flags

* Fix default field

* Fix handling of anyfield (and add "any" as alias)

* Openrewrite...

* More test cases

* Remove non-covered libraries

* Use LIKE syntax as default instead of regex

Disabled exact match

* Update module-info.java

* Create "query" package

* Postgres searcher

* Return back the exact match operator

* checkstyle

* Add link

* WIP

* Fix compilation

* WIP

* Intermediate result

Co-authored-by: Loay Ghreeb <52158423+LoayGhreeb@users.noreply.github.com>

* Query should be OK

Co-authored-by: Loay Ghreeb <52158423+LoayGhreeb@users.noreply.github.com>

* Indexing of split values

Co-authored-by: Loay Ghreeb <52158423+LoayGhreeb@users.noreply.github.com>

* Fix tests compile

* Use first name Last name for authors

* Refactor SQL query visitor

* Adapt tests

* Use join with EXACT_MATCH only

* Update to Postgres 17

* Attempt to use sub-queries with CTEs

* Fix CTEs sub-queries and grouping

TODO: EXCAT_MATCH to search in split table

* group matches by entry_id

* Use NOT IN for negation queries

* Fix unary NOT operator

* Use split values table for EXACT_MATCH queries

* Prepare for linked files index (full-text)

* Prepare linked files tables

* Fix searching

* Use multi column index

* fix merged module issues

* Fix update event

* Remove postgres linked files indexer

* Remove and insert field on update event

* Remove search score column

* Update search groups matches

* Remove search_score from tale preferences

* Migrate search groups flags to new syntax

* Localization

* Fix dialog message

* Ignores groups field from default searches

Fixes https://github.com/JabRef/jabref/issues/7996

* Use TYPE_HEADER field for entrytype

* Search to Lucene query for linked files searching

* Marge linked files and bib fields results

* Searching in background task

* Fix search to SQL tests

* Localization test

* Fix DatabaseSearcherTest

* Fix DatabaseSearcherWithBibFilesTest

* Fix exportMatches test

* Update src/main/java/org/jabref/model/entry/BibEntry.java

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

* Add SINGLE_ENTRY_LINK to latex field

* Remove changelog entries

* Remove groups migration from localization

* Extract search terms from query (ignore negated terms)

* Fix architecture test

* Highlight Preview viewer with Postgres regexp_replace

* OpenRewrite

* Remove onRunning

* Set search query listener in the constructor

* Fix preview tab scrolling

* Use prepared statement to fix escaping

* Use prepared statement for sql query

* Store the start and end positions for every field

* WIP highlight source tab

* Fix source tab highlighting

* Return regex, case-sensitive flags back to the search bar

* Use for search bar flags for unfielded terms

* Skip migrations for unfieleded terms

* Return regex, case-sensitive CheckBox to search groups dialog

* Apply suggestions from code review

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

* Update JabRef_en.properties

* Fix search grammar to support special chars

* Create SearchQueryTest.java

* Adapt SQL visitor with new grammar

* Allow to use quotes without escaping

* escape SQL wildcard chars

* Reorder methods

* Adapt SearchFlagsToExpressionVisitor

* Adapt SearchToLuceneVisitor

* Adapt SearchQueryExtractorVisitor

* Fix tests

* Fix DatabaseSearcherTest

* Fix search terms patten for highlighting

* Highlight source tab field by field according to the search query

* Apply suggestions from code review

* Update src/main/java/org/jabref/gui/importer/actions/SearchGroupsMigrationAction.java

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

* Set default operator to AND

* remove debug

---------

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>
Co-authored-by: Carl Christian Snethlage <50491877+calixtus@users.noreply.github.com>
Co-authored-by: Siedlerchr <siedlerkiller@gmail.com>

Siedlerchr committed:
New Crowdin updates (#12065)

* New translations jabref_en.properties (French)

* New translations jabref_en.properties (Spanish)

* New translations jabref_en.properties (Polish)

* New translations jabref_en.properties (Portuguese, Brazilian)

* New translations jabref_en.properties (Danish)

* New translations jabref_en.properties (German)

* New translations jabref_en.properties (Greek)

* New translations jabref_en.properties (Italian)

* New translations jabref_en.properties (Japanese)

* New translations jabref_en.properties (Korean)

* New translations jabref_en.properties (Dutch)

* New translations jabref_en.properties (Norwegian)

* New translations jabref_en.properties (Portuguese)

* New translations jabref_en.properties (Russian)

* New translations jabref_en.properties (Swedish)

* New translations jabref_en.properties (Turkish)

* New translations jabref_en.properties (Chinese Simplified)

* New translations jabref_en.properties (Vietnamese)

* New translations jabref_en.properties (Indonesian)

* New translations jabref_en.properties (Tagalog)

koppor committed:
Fix link to contribution guide (#12064)

koppor committed:
Fix moving of file into library (#12001)

* Fix comments

* Improve comment

* Fix variable name

* Add list of entries to ImportFilesResultItemViewModel

If any error, no double entries in result set

* Fix support to move files into library

* Handle existing files in the case of "Copy"

* Add link to PR

* Fix mocking for CitationsRelationTabViewModelTest

* Use Files.isSameFile

* Clean filenames at "getValidFileName"

* Add JavaDoc comment

* Rewrite LinkedFileHandler to do the renaming, too

* Remove unused method

* Introduce DragDrop.handleDropOfFiles (to avoid code duplication)

* Group file related methods together (in BibEntry)

* Remove unused methods

* Refine CHANGELOG.md

* Avoid double-addition when linking files

* Fix condition

* Fix flag

* Add notification also on move/copy if file exists

* Fix checkstyle error

* Streamline code

* Fix target directory creation

* Fix compilation, and remove Index Manager from ImportHandler

* Remove IndexManager from EntryEditor

* Update CommentsTabTest.java

---------

Co-authored-by: Loay Ghreeb <loayahmed655@gmail.com>

koppor committed:
Also add message on failing tests (#12063)

* Also add message on failing tests

* Add check for forks

InAnYan committed:
Add template engine ADR (#11900)

* Add template engine ADR

* Update docs/decisions/0039-template-engine.md

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

* Update docs/decisions/0039-template-engine.md

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

* Update docs/decisions/0039-template-engine.md

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

* Update from code review

* Add examples

* Update examples

* Update from code review

* Fix wording

* Add chosen option

* Add chosen option

* Remove blank line

---------

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

koppor committed:
Try github.token

Refs https://github.com/JabRef/jabref/pull/12063

AU0430 committed:
Support RFC links in Quick Import functionality (#11949) (#12043)

* Add RfcId to handle RFC url link to support quick import

* Refactor RFC URL parsing
Add JavaDoc comments
Remove unnecessary lowercase

* Refactor RFC URL parsing
Add JavaDoc comments
Remove unnecessary lowercase

* Adjust code style

* Adjust code style

* According to CI test feedback:
- Removed RFC entry from identifier list(StandardField.java)
- Modified RfcId class to extend EprintIdentifier

* Accorfing to maintainer feedback to fix:
- Replaced RfcId name with RFC class for consistency and clarity.
- Updated RFC parsing logic to support both plain RFC IDs (e.g., "rfc7276") and full URLs (e.g., "https://www.rfc-editor.org/rfc/rfc7276.html").
- Added tests in RFCTest to validate proper parsing of RFC identifiers from both formats.
- Removed unnecessary method (isValid) improved code clarity.
- Adjusted the CompositeIdFetcher to integrate the new RFC class functionalities.

* Refactored RFCTest to adhere to the suggestions provided:
- Converted `testParsePlainRfcId` to a `@ParameterizedTest` using `@CsvSource`.
- Adjusted `testInvalidRfc` to use `assertEquals(Optional.empty(), rfc)`.
- Updated `testGetExternalUri` to verify the URI using `Optional.of()` for consistent checks.

* adjust code format according to CI test(checkstyle)

* Swap expected and input

* Fix OpenRewrite issues

* fix code style according to checkstyle

---------

Co-authored-by: Cheng <u7667316@anu.edu.au>
Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

Boston54 committed:
Caught an error when accessing an invalid path. fixes #10548 (#12038)

* Caught an error when accessing an invalid path. fixes #10548

* Fixed typo in issue num

* Moved changelog entry to unreleased section

* Moved try-catch and added tests

* Fix checkstyle

* Fix checkstyle again

* Fix checkstyle fr this time

* Discard changes to external-libraries.md

* Result of "rewriteRun"

* Add link to issue

* Use VisibleForTesting annotation

* Remove Jimfs depdencency

---------

Co-authored-by: Tom Martin <u7656592@anu.edu.au>
Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

u7465990 committed:
Double click on citation opens LaTeX editor (#12044)

* First version of the double click issue

* First version to use double click to open TeXstudio

* change the click count to two

* I set the GitHub page for the test

* Replace exec() with ProcessBuilder and printStackTrace() with LOGGER

* change some useless functions

* I solve the csl-styles submodel problem

* Solve the avvrv.jabref.org conflict

* The double click function works fine for the emacs external program

* The double click function works fine for the TeXworks external program

* I changed the required code. I also delete some useless comment. I make TeXworks works for this feature

* finished texWorks and delete some comments

* delete some comments

* solve untrack files

* I make Sublime Text works for this feature

* I make WinEdt works for this feature

* delete some command and change jumpString to jumpToLineCommandlineArguments

* Fix csl-styles
  Please enter the commit message for your changes. Lines starting

* Fix abbrv.jabref.org submodules

* I make Vim works for this feature

* I rewrite jumpToLine for TexShop. But I don't hava a Mac OS machine. I hope it's working.

* Delete the JumpToLine in TeXshop and a comment line in TeXstudio

* Fix submodules (and wrong file)

* Fix empty lines

* Default should be TeXstudio

* Remvoe "Integer.format" (and add test)

* Fix PushToVim

* Cleanup PushToLyx

* Refactor PushToVim

* Add CHANGELOG.md entry

---------

Co-authored-by: Oliver Kopp <kopp.dev@gmail.com>

koppor committed:
Identifier: Rename "getNormalized" to "asString" (#12054)

* Identifier: Rename "getNormalized" to "asString" (and replace getDOI by asString)

* Fix method name

* Fix checkstyle

* Fix method name

* Fix methodname ... agian ...

* Adapt to new method names

* Fix method name ... again ...

Siedlerchr committed:
Change casing and improve naming of Background tasks (#12066)

Fixes

tomeg09 committed:
WIP: Rename General File Directory to Library-Specific File Directory (#12037)

* Change enum & localisation preferences from "General File Directory" to "Library File Directory"

* change 'generalFileDirectory' to 'libraryFileDirectory' in codebase

* change 'generalFileDirectory' to 'libraryFileDirectory' in test code

* change 'General file directory' to 'Library file directory' in places that were previously missed.

* amendment to prior commits: should show up in GUI as "Library-specific File Directory". Have followed convention for "User-Specific File Directory" where this is referred to outside of UI prompts as "Library File Directory"

* update CHANGELOG.md to document activity in this branch.

* Additional/missed locations: change 'General File Directory' to 'Library-specific File Directory'

* Per Koppor's instructions: unify front/back end to "Library-specific File Directory"

* update code comments to reflect change from library file directory to library-specific file directory

* updated en localisation per unit tests:
1. removed unused general file directory key
2. made Library-specific file directory key all-lowercase per unit test requirements.

---------

Co-authored-by: rl712 <176219446+rl712@users.noreply.github.com>

koppor committed:
Disable meta data elements in adr-template (#12074)

Siedlerchr committed:
New Crowdin updates (#12075)

juliusalberto committed:
Merge branch 'main' of https://github.com/JabRef/jabref into fix-issue-warning-message

juliusalberto committed:
Implemented suggestions to use collectors, removed unused comments and methods
